### PR TITLE
Add missing returns.

### PR DESCRIPTION
**Pull request by pq**
As of `1.18.0-dev-0`, these cases will get flagged.

### CODE REVIEW

File: packages/flutter/lib/src/material/icon.dart [hunk 1]
```diff
@@ -65,6 +65,7 @@ class Icon extends StatelessWidget {
       case Brightness.light:
         return Colors.black;
     }
+    return null;
```
Comment (abarth) [inline packages/flutter/lib/src/material/icon.dart]: Why do we need a return null in this case?  We've handled all the enum values in the switch.  If the types are correct, this code is dead.

**Comment by pq** [comment]
cc @abarth @Hixie 

This will make the PR tomorrow when there's a fresh dev build to pull in a little more manageable...

File: packages/flutter/lib/src/material/icon.dart (hunk 1 above)
Comment (pq) [inline packages/flutter/lib/src/material/icon.dart]: `brightness` could be `null`.

**Comment by Hixie** [comment]
We should assert(foo != null); in all the cases that come after a switch that covers every case.

File: examples/flutter_gallery/lib/demo/calculator/logic.dart [hunk 2]
```diff
@@ -258,6 +260,7 @@ class CalcExpression {
       case ExpressionState.Result:
         return appendOperation(Operation.Subtraction);
     }
+    return null;
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/calculator/logic.dart]: I'd make this one a `default:` case, i think, to make it more obvious it was intended to catch the other cases. Actually I'm surprised that this compiles, I guess maybe we currently don't have any cases where you can't append a `-` ?

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart [hunk 3]
```diff
@@ -46,7 +46,7 @@ class GridDemoPhotoItem extends StatelessWidget {
   }
 
   final Photo photo;
-  final GridDemoTileStyle tileStyle;
+final GridDemoTileStyle tileStyle;
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: typo

**Comment by pq** [comment]
@Hixie : added asserts.  Thinking I got 'em all...  Assuming build is green, PTAL.

File: packages/flutter/lib/src/material/raised_button.dart [hunk 4]
```diff
@@ -98,12 +98,15 @@ class RaisedButton extends StatelessWidget {
     } else {
       if (disabledColor != null)
         return disabledColor;
+      Brightness brightness = Theme.of(context).brightness;
       switch (Theme.of(context).brightness) {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/raised_button.dart]: you missed updating the switch statement expression here

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/raised_button.dart (hunk 4 above)
Comment (pq) [inline packages/flutter/lib/src/material/raised_button.dart]: Ah!  Can you tell I'm weary? ;) 

Thanks!

File: examples/flutter_gallery/lib/demo/calculator/logic.dart (hunk 2 above)
Comment (pq) [inline examples/flutter_gallery/lib/demo/calculator/logic.dart]: Done!

**Pull request merged**